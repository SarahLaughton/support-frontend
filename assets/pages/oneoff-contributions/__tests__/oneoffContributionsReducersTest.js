// @flow

// ----- Imports ----- //

import type { CheckoutFailureReason } from 'helpers/checkoutErrors';
import createReducer from '../oneOffContributionsReducer';


// ----- Tests ----- //

describe('One-off Reducer', () => {

  const reducer = createReducer(20, 'GBPCountries');

  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle CHECKOUT_ERROR', () => {

    const insufficientFunds: CheckoutFailureReason = 'insufficient_funds';
    const action = {
      type: 'CHECKOUT_ERROR',
      checkoutFailureReason: insufficientFunds,
    };

    const newState = reducer(undefined, action);

    expect(newState.oneoffContrib.checkoutFailureReason).toEqual(insufficientFunds);
  });

  it('should handle SET_PAYPAL_BUTTON', () => {

    const value = 'ExpressCheckout';
    const action = {
      type: 'SET_PAYPAL_BUTTON',
      value,
    };

    const newState = reducer(undefined, action);

    expect(newState.oneoffContrib.checkoutFailureReason).toMatchSnapshot();
  });

});
