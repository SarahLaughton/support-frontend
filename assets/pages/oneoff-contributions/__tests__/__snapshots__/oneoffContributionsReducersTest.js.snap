// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`One-off Reducer should handle SET_PAYPAL_BUTTON 1`] = `null`;

exports[`One-off Reducer should return the initial state 1`] = `
Object {
  "csrf": Object {
    "token": null,
  },
  "marketingConsent": Object {
    "confirmOptIn": null,
    "error": false,
  },
  "oneoffContrib": Object {
    "amount": 20,
    "error": null,
    "paymentComplete": false,
  },
  "user": Object {
    "displayName": "",
    "email": Object {
      "shouldValidate": false,
      "value": "",
    },
    "firstName": Object {
      "shouldValidate": false,
      "value": "",
    },
    "fullName": Object {
      "shouldValidate": false,
      "value": "",
    },
    "gnmMarketing": false,
    "id": "",
    "isPostDeploymentTestUser": false,
    "isSignedIn": false,
    "isTestUser": null,
    "lastName": Object {
      "shouldValidate": false,
      "value": "",
    },
  },
}
`;
